datafusion/expr/src/utils.rs

```diff
@@ -1107,20 +1107,49 @@ fn split_binary_impl<'a>(
 /// assert_eq!(conjunction(split), Some(expr));
 /// ```
 pub fn conjunction(filters: impl IntoIterator<Item = Expr>) -> Option<Expr> {
-    filters.into_iter().reduce(|accum, expr| accum.and(expr))
+    filters.into_iter().reduce(Expr::and)
```

Comment:
this is the very minor simplification and while I was in here I figured I would improve the docs too